### ISSUE TITLE
Master server doesn't support IPv6

### ISSUE THREAD

**Issue by ccoors** [OpenRCT2/OpenRCT2#4953]
**OS:** Arch Linux
**Version:** 0.0.6 develop and 0.0.5 release
**Commit/Build:** 99a662f

I'm trying to advertise a game and I get this error:
> Unable to advertise: An unknown error has occured. If this state persists, please contact the developers.

- [ ] Reproducible in RCT2 (vanilla)?
- [x] Multiplayer?

**Steps to reproduce:**
1. Edit your config.ini to some reasonable settings
2. Run OpenRCT2 headless server
> $ openrct2 host ~/data/archive/Software/RCT2/scenarios/Simple\ Sandbox.sc6 --port 11753 --headless --verbose
[...]
VERBOSE: begin openrct2 loop
Unable to advertise: An unknown error has occured. If this state persists, please contact the developers.

I found this in NetworkServerAdvertiser.cpp:177
```c++
const char * message = "Invalid response from server";
json_t * jsonMessage = json_object_get(jsonRoot, "message");
if (json_is_string(jsonMessage))
{
    message = json_string_value(jsonMessage);
}
Console::Error::WriteLine("Unable to advertise: %s", message);
```
For debugging I added
```c++
char * jsonDump = json_dumps(jsonRoot, 0);
Console::Error::WriteLine("%s", jsonDump);
free(jsonDump);
```
which gives me

> {"status": 500, "message": "An unknown error has occured. If this state persists, please contact the developers."}

on the console, which is not really helpful.

My config.ini:
[config.ini.txt](https://github.com/OpenRCT2/OpenRCT2/files/678571/config.ini.txt)

**Comment by IntelOrca** [comment]
An error is happening on the master server, I am not sure why as I don't maintain it. I did try hosting my own server though and it successfully advertised, so I expect there is something wrong with your configuration.

The think the master server only supports IPv4.

**Comment by ccoors** [comment]
Yes, you are right, if I disable IPv6 locally it works. Though that's not really a solution. I'd look into it, but I'm not familiar with Node.js.

**Comment by janisozaur** [comment]
https://github.com/OpenRCT2/OpenRCT2/blob/99a662f/src/network/NetworkServerAdvertiser.cpp#L103 `.size` is not set, but used in `http_request_async`, same in `SendHeartbeat`. I don't know what it hould be set to, but looks like it could cause malformed request.

**Comment by tian2992** [comment]
I can report the same issue on a dual stacked IPv4 + IPv6 host. Unable to advertise.

**Comment by janisozaur** [comment]
There is more discussion/investigation in #6277 and a pending fix in #7103.

**Comment by janisozaur** [comment]
Fixed in #7103
